Sources/Confidence/ConfidenceValue.swift

```diff
@@ -0,0 +1,231 @@
+import Foundation
+
+public typealias ConfidenceStruct = [String: ConfidenceValue]
+
+public class ConfidenceValue: Equatable, Encodable {
+    private let value: ConfidenceValueInternal
+
+    public init(boolean: Bool) {
+        self.value = .boolean(boolean)
+    }
+
+    public init(string: String) {
+        self.value = .string(string)
+    }
+
+    public init(integer: Int64) {
+        self.value = .integer(integer)
+    }
+
+    public init(double: Double) {
+        self.value = .double(double)
+    }
+
+    public init(date: DateComponents) {
+        self.value = .date(date)
+    }
+
+    public init(timestamp: Date) {
+        self.value = .timestamp(timestamp)
+    }
+
+    // TODO: Handle heterogeneous types
+    public init(valueList: [ConfidenceValue]) {
```

Comment:
So, either we hide(private) this completely or we do some work to validate that the list contains heterogeneous types. 

My main question is: What do we do if it _doesn't_? Do we throw? Do we make a best guess on what is "correct" and drop incorrect types?
Unless we can find good answers to those questions I think we should make it private.

Reply:
We should hide it because list in events doesn't accept heterogenous types. I added best-effort code to convert heterogenous lists from OF Values, that defaults to null values